### PR TITLE
Fixing the Body Archive as well as a whole lot of bullshit

### LINKED ISSUE #34178: Body Archive Failing
The WHOLE POINT of the body archive is to have a backup in case someone gets gibbed or sharded and yet if this happens it spawns them as a brain mob instead. Disaster!

Also keep clothes seems to do nothing all(?) of the time
Also, SPAWN+TRANSFER often doesn't seem to actually transfer their mind

### LINKED ISSUE #34425: I DID BUT CAN RESPAWN PART 2
(WEB REPORT BY: loveender REMOTE: 206.221.180.138:7777)
# Revision
684c5eb98f4006c575b52cd61003e0c32bce99ed

# Description
Follow up from previous bug report
# Steps to Reproduce
So I obsed instead of joining after I had the option to rejoin the game from the previous crash.
I took a gamble and clicked the red xenoarch stone that spits out clones of whoever touches it.
Anyway clicked it, game crashed again.
Reloaded and I'm able to rejoin the game again

# What you Expected

# What Actually Happened

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #34178
Fixes #34425 (maybe?)

image

https://github.com/vgstation-coders/vgstation13/assets/7573912/518cdee2-5489-479d-a5e6-285f3890f87f

Fixes various issues. Some brought by Kanef who didn't test their Body Archive changes, some by Ada who fixed Midround Wizards in the worst way possible (treating the symptoms instead of the cause), and some by older cursed code.

It's almost all backend changes:
* The ticker.minds list now actually properly fills up with all created minds, instead of only a few random ones.
* Reverted ada's changes to mind.transfer_to() which caused a new mind to be created each time and immediately discarded, clogging up the Body Archive with useless crap.
* the one_click_antag.dm's makeBody() proc no longer borks the minds of its mobs, fixing several potential issues notably with midround antags that relied on this proc (aka the real reason why midround Wizards ghosted themselves on self-SoC)
* Fixed (and partially redesigned) Kanef's body archive outfiter. You can now spawn clothed copies with a separate button. Body Archives also now keep track of their origin mind allowing the Body Archive Panel to transfer them to any new mob the admins desire (it's pretty amazing actually, you can transfer a player's mind to a new mob even if they disconnected, and they'll be inside that new mob upon reconnecting).
* A bit of cleanup and copypasta removal

### CODE REVIEW

File: code/datums/body_archive.dm [hunk 1]
```diff
@@ -10,6 +10,8 @@ var/list/body_archives = list()
 	var/key
 	var/rank
 
+	var/datum/mind/mind//keeping a pointer to the source mind for tracking purposes
```
Comment (Exxion) [inline code/datums/body_archive.dm]: The actual compiler seems to deal with this just fine, but the syntax highlighting doesn't like it

File: code/modules/admin/verbs/one_click_antag.dm [hunk 2]
```diff
@@ -229,10 +229,6 @@
 	new_character.dna.ready_dna(new_character)
 	new_character.key = G_found.key
 
-	// Create a brand new mind for the dude
-	new_character.mind = new
-	new_character.mind.current = new_character
-
```
Comment (DeityLink) [inline code/modules/admin/verbs/one_click_antag.dm:235]: This caused mobs created with makeBody() to spawn with an uninitialized mind, leading to several more issues such as #33027

File: code/datums/gamemode/role/role.dm [hunk 3]
```diff
@@ -145,9 +145,6 @@
 		return 0
 
 	antag = M
-	if(M.current.client)
-		antag.key = M.current.client.ckey
-		antag.active = TRUE
```
Comment (DeityLink) [inline code/datums/gamemode/role/role.dm:150]: Removing the symptom treatment. The cause came from one_click_antag.dm's makeBody().

**Comment by Exxion** [comment]
Probably fixes #34425, since what I did to fix it in testing was basically one of the things done in here

**Pull request merged**